console/v1/types_console_yaml_sample.go

```diff
@@ -0,0 +1,26 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleYAMLSample is an extension for customizing OpenShift web console YAML samples.
+type ConsoleYAMLSample struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ConsoleYAMLSampleSpec `json:"spec"`
+}
+
+// ConsoleYAMLSampleSpec is the desired YAML sample configuration.
+type ConsoleYAMLSampleSpec struct {
+	metav1.TypeMeta `json:",inline"`
+	// title of the YAML sample.
+	Title string `json:"title"`
```

Comment:
We'll want regex patterns for each of these strings to make sure they aren't empty strings

Reply:
wont it be enough to have these fields as `required` in the CRD ?

Reply:
No, the empty string is allowed even for required fields